File: Maths Problems/findNumbers.py
'''
https://leetcode.com/problems/find-numbers-with-even-number-of-digits/
'''

class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        evens = 0 
        for num in nums:
            if len(str(num))%2 == 0:
                evens += 1
        return evens

class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        evens = 0 
        for num in nums:
            size = 0 
            while (num > 1):
                num /= 10
                size += 1
            if (size % 2 == 0) and (size > 0):
                evens += 1
        return evens

#From LC discussions
class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        return sum(int(math.log10(n)) % 2 for n in nums)

File: Maths Problems/findNumbers.cpp
#include <math.h>

//The rolled out cpp version of my list comprehension python solution
class Solution {
public:
    int findNumbers(vector<int>& nums) {
        int count = 0;
        for (int i = 0; i < nums.size(); i++){
            count += (int)(log10(nums[i])) % 2;
            }
        return count;
    }
};